### PR TITLE
Support bigquery `CAST AS x [STRING|DATE] FORMAT` syntax

### PR DESCRIPTION
**Pull request by lustefaniak**
https://cloud.google.com/bigquery/docs/reference/standard-sql/conversion_functions#cast_as_date

### CODE REVIEW

File: src/ast/mod.rs [hunk 1]
```diff
@@ -437,19 +437,22 @@ pub enum Expr {
     Cast {
         expr: Box<Expr>,
         data_type: DataType,
+        format: Option<Value>,
```
Comment (alamb) [inline src/ast/mod.rs:440]: can you please update the documentation here to explain what `format` is (aka what SQL it comes from)?

File: src/ast/mod.rs [hunk 2]
```diff
@@ -437,19 +437,22 @@ pub enum Expr {
...
     },
     /// TRY_CAST an expression to a different data type e.g. `TRY_CAST(foo AS VARCHAR(123))`
     //  this differs from CAST in the choice of how to implement invalid conversions
     TryCast {
         expr: Box<Expr>,
         data_type: DataType,
+        format: Option<Value>,
```
Comment (alamb) [inline src/ast/mod.rs:447]: FWIW it appears bigquery supports a more sophisticated format clause than just a single value:
https://cloud.google.com/bigquery/docs/reference/standard-sql/format-elements#formatting_syntax

Perhaps we should create a `CastFormat` enum so if we want to extend the supported syntax the API change is less. Something like:

```suggestion
        format: Option<CastFormat>,
```

```rust
/// Options for `CAST` / `TRY_CAST` 
/// BigQuery: https://cloud.google.com/bigquery/docs/reference/standard-sql/format-elements#formatting_syntax
enum CastFormat {
   /// Constant string value
  String(Value)
 }
```

**Review by alamb — COMMENTED** [review]
Thank you @lustefaniak  -- sorry I missed this one in the sea of reiviews. I think it is looking good in my mind 👍

File: src/ast/mod.rs (hunk 2 above)
Comment (alamb) [reply src/ast/mod.rs:447]: (I don't feel strongly about this -- we can merge it as is as well)

**Comment by alamb** [comment]
I think the CI failures are not related to the changes in this PR -- if you merge up / rebase from main they should be solved. See https://github.com/sqlparser-rs/sqlparser-rs/pull/995

File: src/ast/mod.rs (hunk 1 above)
Comment (lustefaniak) [reply src/ast/mod.rs]: Changed

File: src/ast/mod.rs (hunk 2 above)
Comment (lustefaniak) [reply src/ast/mod.rs]: I think this is really good idea, I added CastFormat and covered `AT TIME ZONE` support too.

**Review by alamb — APPROVED** [review]
Looks great -- thank you @lustefaniak

**Pull request merged**